### DISCUSSION TITLE
Hierarchy of Geo location and Announcement to specific geo location hierarchy

### DISCUSSION THREAD

**Discussion by manjirisulekavadi** [Sunbird-Ed/Community#264] [Documentation] [answered]
Use case
- Having org units at various geographical hierarchical locations like states, districts, blocks, clusters, city/village etc.
- Need to announce batches or other purpose announcement restricted to any level of geographical hierarchy (for all district level batched only, etc.)

1. Can we associate multiple hierarchical Geo Locations to a tenant?
2. Can we configure and sent announcement to any Geo Location Level or Unit?    
3. Links to Geo Location APIs, Announcement APIs are broken on http://docs.sunbird.org/latest/developer-docs/configuring_sunbird/configuring_announcement/ and get 404 error

**Comment by Krishnaj20** [comment, marked as answer]
@manjirisulekavadi - are you talking about notifications here? Please clarify. 
I recommend you to refer to the latest documentation on the microsites. You can find the links to them [here](https://sunbird.org/building-blocks). Link to the notification service documentation from Lern microsite is [here](https://lern.sunbird.org/learn/product-and-developer-guide/notification-service). 

@keshavprasadms @rhwarrier

**Comment by manjirisulekavadi** [comment]
@Krishnaj20 Yes the Location APIs link to which you directed me gives parent-child code and enables hierarchy. 

I also went through the notification APIs.  I had referred to [here] (http://docs.sunbird.org/2.7.0/developer-docs/configuring_sunbird/configuring_announcement/) in the old documents for organizations to announce events, courses, news, circulars, etc. to all or select members, in all or some of the organization’s operative locations.

It mentioned, "The Announcements feature does not support location hierarchy. While targeting an announcement, all locations are available in a flat structure irrespective of location type in the database."

-------------

Have the notification APIs made the Announcement APIs obsolete?

**Reply by Krishnaj20** [reply]
@reshmi-nair - can you confirm whether the announcement APIs are now sunset and are now part of notification service?

@vrayulu @rhwarrier
